File: DormService/Services/Laundry/Laundry.API/Entities/WashingMachine.cs
using MongoDB.Bson;

namespace Laundry.API.Entities;

public class WashingMachine
{

    public string _id { get; set; }
    public string ConfigurationId { get; set; }
    public string? Time { get; set; }
    public string? Date { get; set; }
    public bool Reserved { get; set; } 
    public int SpinRate { get; set; }
    public int WashingTemperature { get; set; }
    

    public WashingMachine(string id, string date, string time) 
    {
        _id = ObjectId.GenerateNewId().ToString();
        ConfigurationId = id;
        Date = date;
        Time = time;
        Reserved = false;
        SpinRate = 0;
        WashingTemperature = 0;
    }
}

File: DormService/DormApps/dorm-spa/src/Apps/LaundryPage/models/WashingMachine.ts
export type WashingMachine = {
    _id: string;
    id?: string;
    configurationId: string;
    studentId: string;
    time: string;
    date: string;
    reserved: boolean;
    spinRate?: number;
    washingTemperature?: number;
};

export type WashingMachineConfigurations = {
    _id: string;
    manufacturer: string;
    startDate: string;
    expirationDate: string;
    utilizationFactor: number;
    responsiblePersonEmail: string;
}

export type WashingMachineConfigurationRequest = {
    manufacturer?: string;
    startDate?: string;
    expirationDate?: string;
    responsiblePersonEmail?: string;
}

export enum Timeframes {
    MORNING = "Morning",
    NOON = "Noon",
    AFTERNOON = "Afternoon"
}